### PR TITLE
feat(  AM-5594): Updating kubeslice webhook to work with new nsm

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

File: pkg/webhook/pod/webhook.go [hunk 1]
```diff
@@ -102,7 +103,7 @@ func Mutate(pod *corev1.Pod, sliceName string) *corev1.Pod {
 
 	// Add vl3 annotation to pod template
 	annotations := pod.ObjectMeta.Annotations
-	annotations[nsmInjectAnnotaionKey] = "vl3-service-" + sliceName
+	annotations[nsmInjectAnnotaionKey] = fmt.Sprintf("kernel://vl3-service-%s/nsm0", sliceName)
```
Comment (kmjayadeep) [inline pkg/webhook/pod/webhook.go:106]: does this work only with kernel forwarder? or will it work with vpp as well?

**Comment by kmjayadeep** [comment]
can you add some UT for checking the annotations, if there isn;t one already?

File: pkg/webhook/pod/webhook.go (hunk 1 above)
Comment (mohammedzee1000) [reply pkg/webhook/pod/webhook.go:106]: I guess for now atleast our goal is the kernel forwarder. When we support other forwarders, we can look if we need to change this.

**Comment by mohammedzee1000** [comment]
> can you add some UT for checking the annotations, if there isn;t one already?

As the checks have gone green, i am assuming rhere arent any. So I would say we need to add one

File: pkg/webhook/pod/webhook.go [hunk 2]
```diff
@@ -39,7 +40,8 @@ const (
 	PodInjectLabelKey                         = "kubeslice.io/pod-type"
 	admissionWebhookSliceNamespaceSelectorKey = controllers.ApplicationNamespaceSelectorLabelKey
 	controlPlaneNamespace                     = "kubeslice-system"
-	nsmInjectAnnotaionKey                     = "ns.networkservicemesh.io"
+	nsmInjectAnnotaionKey1                    = "ns.networkservicemesh.io"
```
Comment (bharath-avesha) [inline pkg/webhook/pod/webhook.go:43]: Please fix the spelling of "Annotation".

**Pull request closed without merging**